src/StatusQ/Controls/StatusBaseButton.qml

```diff
@@ -19,131 +20,110 @@ Rectangle {
+    property color disabledColor
+
+    property color textColor
+    property color disabledTextColor
+    property color borderColor: "transparent"
 
     property int size: StatusBaseButton.Size.Large
     property int type: StatusBaseButton.Type.Normal
 
-    property alias text: label.text
-    property alias font: label.font
-
-    property real defaultLeftPadding: size === StatusBaseButton.Size.Large ? 24 : 12
-    property real defaultRightPadding: size === StatusBaseButton.Size.Large ? 24 : 12
-    property real defaultTopPadding: {
-        switch (size) {
-            case StatusBaseButton.Size.Tiny:
-              return 5
-            case StatusBaseButton.Size.Small:
-              return 10
-            case StatusBaseButton.Size.Large:
-            default:
-              return 11
-        }
-    }
-    property real defaultBottomPadding: {
-        switch (size) {
+    property int defaultHorizontalPadding: (horizontalPadding > 0) ? horizontalPadding : ((size === StatusBaseButton.Size.Large) ? 24 : 12)
+    property int defaultVerticalPadding: {
+        if (verticalPadding > 0) {
+            verticalPadding;
+        } else {
+            switch (size) {
             case StatusBaseButton.Size.Tiny:
-              return 5
+                return 5;
             case StatusBaseButton.Size.Small:
-              return 10
+                return 10;
             case StatusBaseButton.Size.Large:
             default:
-              return 11
+                return 11;
+            }
         }
     }
 
-
-    property real leftPadding: defaultLeftPadding
-    property real rightPadding: defaultRightPadding
-    property real topPadding: defaultTopPadding
-    property real bottomPadding: defaultBottomPadding
-
-    property color normalColor
-    property color hoverColor
-    property color disaledColor
-
-    property color textColor
-    property color disabledTextColor
-
-    signal pressed(var mouse)
-    signal released(var mouse)
-    signal clicked(var mouse)
-    signal pressAndHold(var mouse)
-
-    property bool highlighted: false
-
-
-    /// Implementation
-    implicitWidth: layout.width + statusBaseButton.leftPadding + statusBaseButton.rightPadding
-    implicitHeight: layout.height + statusBaseButton.topPadding + statusBaseButton.bottomPadding
-
-
-    radius: size !== StatusBaseButton.Size.Tiny ? 8 : 6
-
-    color: {
-        if (statusBaseButton.enabled)
-            return sensor.containsMouse || highlighted ? hoverColor
-                                                       : normalColor;
-        return disaledColor
+    implicitWidth: (contentItem.width + root.leftPadding + root.rightPadding)
+    implicitHeight: (contentItem.height + root.topPadding + root.bottomPadding)
+    verticalPadding: 0
+    horizontalPadding: 0
+    leftPadding: defaultHorizontalPadding
+    rightPadding: defaultHorizontalPadding
+    topPadding: defaultVerticalPadding
+    bottomPadding: defaultVerticalPadding
```

Comment:
You don't need these 4 paddings

Reply:
regarding paddings, for example we could leave it as:
```
    leftPadding: size === StatusBaseButton.Size.Large ? 24 : 12
    rightPadding: size === StatusBaseButton.Size.Large ? 24 : 12
    bottomPadding: {
        switch (size) {
        case StatusBaseButton.Size.Tiny:
            return 5;
        case StatusBaseButton.Size.Small:
            return 10;
        case StatusBaseButton.Size.Large:
        default:
            return 11;
        }
    }
    topPadding: {
        switch (size) {
        case StatusBaseButton.Size.Tiny:
            return 5;
        case StatusBaseButton.Size.Small:
            return 10;
        case StatusBaseButton.Size.Large:
        default:
            return 11;
        }
    }

    implicitWidth: (contentItem.width + root.leftPadding + root.rightPadding)
    implicitHeight: (contentItem.height + root.topPadding + root.bottomPadding)
```
but then if the user sets horizontal or vertical paddings, nothing changes.

Reply:
Yeah but since left/right and top/bottom are the same, you could specify horizontalPadding and verticalPadding only